src/app/application/components/route-view/RouteView.vue

```diff
@@ -91,17 +79,72 @@ export type RouteView = typeof routeView
 const props = defineProps({
   name: {
     type: String,
-    // once we have rolled out naming everywhere we can change this to required
-    required: false,
-    default: '',
+    required: true,
   },
   attrs: {
     type: Object,
     required: false,
     default: () => ({}),
   },
+  params: {
+    type: Object,
+    required: false,
+    default: () => ({}),
+  },
+})
+
+const routeParams = computed(() => {
+  const params = Object.fromEntries(Object.entries({
+    ...props.params,
+    ...route.params,
+    ...route.query,
+  }).map(([prop, value]) => {
+    let param = urlParam(value)
+    const def = props.params[prop]
+    switch (true) {
+      // if the defined param is a number check to see that the query param
+      // one can pass as a number and if not provide the default instead
+      case typeof def === 'number':
+        if (isNaN(Number(value))) {
+          param = String(def)
+        }
+        break
+    }
+    return [
+      prop,
+      decodeURIComponent(param),
+    ]
+  }))
+  return params
 })
 
+let newParams = {}
+const routerPush = beforePaint((params: Record<string, string | undefined>) => {
+  router.push({
+    name: props.name,
+    query: cleanQuery(params, route.query),
+  })
+  newParams = {}
+})
+const routeUpdate = (params: Record<string, string | undefined>) => {
+  newParams = {
+    ...newParams,
+    ...params,
+  }
+  routerPush(newParams)
+}
```

Comment:
We "save" any attempts to `router.push` here and call the actual `router.push` in a raf-like thing so we don't lose any updates. I guess we'll want to wipe some out/unset things at somepoint, if we do we can use `undefined` to unset things.